client-app/src/examples/contact/detail/DetailsPanel.js

```diff
@@ -0,0 +1,116 @@
+import {textArea, textInput} from '@xh/hoist/desktop/cmp/input';
+import {button} from '@xh/hoist/desktop/cmp/button/index';
+
+export const detailsPanel = hoistCmp.factory({
+    model: uses(DetailsPanelModel),
+    className: 'contact-details-panel',
+
+    render({model, className, ...props}) {
+        const {currentRecord} = model;
+
+        return panel({
+            title: currentRecord?.data.name ?? 'Select a contact',
+            className,
+            icon: Icon.detail(),
+            width: 325,
+            flex: 'none',
+            items: currentRecord ? renameThisProfile({record: currentRecord}) : placeholder('Select a contact to view their details.'),
+            ...props
+        });
+    }
+});
+
+const renameThisProfile = hoistCmp.factory({
+    render({model, record}) {
+        const {data} = record;
+        const {profilePicture, isFavorite} = data;
+        const readonlyRenderer = (val) => val ?? '-';
+
+        return div({
+            className: 'contact-details-panel__inner',
+            items: [
+                profilePicture ? img({
+                    src: profilePicture,
+                    className:
+                        'contact-record-user-image'
+                }) : Icon.user({
+                    size: '10x',
+                    className: 'contact-record-user-image'
+                }),
+                form({
+                    fieldDefaults: {
+                        inline: true,
+                        labelWidth: 100
+                    },
+                    items: [
+                        formField({
+                            field: 'name',
+                            readonlyRenderer,
+                            item: textInput()
+                        }),
+                        formField({
+                            field: 'email',
+                            readonlyRenderer,
+                            item: textInput()
+                        }),
+                        formField({
+                            field: 'location',
+                            readonlyRenderer,
+                            item: textInput()
+                        }),
+                        formField({
+                            field: 'workPhone',
+                            readonlyRenderer,
+                            item: textInput()
+                        }),
+                        formField({
+                            field: 'cellPhone',
+                            readonlyRenderer,
+                            item: textInput()
+                        }),
+                        formField({
+                            field: 'homePhone',
+                            readonlyRenderer,
+                            item: textInput()
+                        }),
+                        formField({
+                            field: 'bio',
+                            readonlyRenderer,
+                            item: textArea()
```

Comment:
Whenever I have a long series of configs like this with a fair amount of duplication, I look to DRY up by spreading a partial with the defaults - e.g.

```
const ffConf = {
  readonlyRenderer: (val) => val ?? '-', 
  item: textInput()
};
```

Then you can really tighten up these formFields:

```
items: [
    formField({...ffConf, field: 'name'}),
    formField({...ffConf, field: 'email'}),
    formField({...ffConf, field: 'location'}),
    formField({...ffConf, field: 'workPhone'}),
    formField({...ffConf, field: 'cellPhone'}),
    formField({...ffConf, field: 'homePhone'}),
    formField({...ffConf, field: 'bio', item: textArea()})
]
```